modules/platforms/cpp/core-test/src/cache_query_schema_test.cpp

```diff
@@ -0,0 +1,311 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdint.h>
+
+#include <iterator>
+
+#include <boost/test/unit_test.hpp>
+
+#include <ignite/common/utils.h>
+
+#include "ignite/cache/cache.h"
+#include "ignite/cache/query/query_cursor.h"
+#include "ignite/cache/query/query_sql_fields.h"
+#include "ignite/ignite.h"
+#include "ignite/ignition.h"
+#include "ignite/test_utils.h"
+
+using namespace boost::unit_test;
+
+using namespace ignite;
+using namespace ignite::cache;
+using namespace ignite::cache::query;
+using namespace ignite::common;
+
+using ignite::impl::binary::BinaryUtils;
+
+/**
+ * Ensure that cursor is empy fails.
+ *
+ * @param cur Cursor.
+ */
+template<typename Cursor>
+void CheckCursorEmpty(Cursor& cur)
```

Comment:
In .NET code base we are starting to adapt more descriptive test names, maybe we can do that for C++ too.
In this case it can be something like `ChechEmptyCursorGetNextThrowsException`